adapter/pkg/subscription/subscription.go

```diff
@@ -191,22 +225,55 @@ func invokeService(endpoint string, responseType interface{}, c chan response, r
 	resp, err := client.Do(req)
 
 	if err != nil {
-		c <- response{err, nil, endpoint, responseType}
+		c <- response{err, nil, endpoint, gatewayLabel, responseType}
 		logger.LoggerSubscription.Errorf("Error occurred while calling the REST API: "+serviceURL, err)
 		return
 	}
 
 	responseBytes, err := ioutil.ReadAll(resp.Body)
 	if resp.StatusCode == http.StatusOK {
 		if err != nil {
-			c <- response{err, nil, endpoint, responseType}
+			c <- response{err, nil, endpoint, gatewayLabel, responseType}
 			logger.LoggerSubscription.Errorf("Error occurred while reading the response received for: "+serviceURL, err)
 			return
 		}
-		c <- response{nil, responseBytes, endpoint, responseType}
-
+		logger.LoggerSubscription.Debug("the request to the control plane over the REST API: " + serviceURL + " is successful.")
+		c <- response{nil, responseBytes, endpoint, gatewayLabel, responseType}
 	} else {
-		c <- response{errors.New(string(responseBytes)), nil, endpoint, responseType}
+		c <- response{errors.New(string(responseBytes)), nil, endpoint, gatewayLabel, responseType}
 		logger.LoggerSubscription.Errorf("Failed to fetch data! "+serviceURL+" responded with "+strconv.Itoa(resp.StatusCode), err)
 	}
 }
+
+func retrieveAPIListFromChannel(c chan response) {
+	for response := range c {
+		responseType := reflect.TypeOf(response.Type).Elem()
+		newResponse := reflect.New(responseType).Interface()
+
+		if response.Error == nil && response.Payload != nil {
+			err := json.Unmarshal(response.Payload, &newResponse)
+
+			if err != nil {
+				logger.LoggerSubscription.Errorf("Error occurred while unmarshalling the APIList response received for: "+response.Endpoint, err)
```

Comment:
```suggestion
				logger.LoggerSubscription.Errorf("Error occurred while unmarshalling the APIList response received for: " + response.Endpoint, err)
```

Reply:
VSCode plugin however removes the space between when I save.